lom/src/lib/lomcommon/yang_parser/main.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+    "fmt"
+    "lom/src/lib/lomcommon/yang_parser/yang_utils"
+    "os"
+)
+
+const (
+    device_health_actions_configs           string = "device-health-actions-configs"
+    device_health_actions_configs_file_name string = "device-health-actions-configs.yang"
+
+    device_health_bindings_configs           string = "device-health-bindings-configs"
+    device_health_bindings_configs_file_name string = "device-health-bindings-configs.yang"
+
+    device_health_globals_configs           string = "device-health-global-configs"
+    device_health_globals_configs_file_name string = "device-health-global-configs.yang"
+
+    device_health_procs_configs           string = "device-health-procs-configs"
+    device_health_procs_configs_file_name string = "device-health-procs-configs.yang"
+
+    actions_conf_file_name  string = "actions.conf.json"
+    bindings_conf_file_name string = "bindings.conf.json"
+    globals_conf_file_name  string = "globals.conf.json"
+    procs_conf_file_name    string = "procs.conf.json"
+)
+
+func main() {
```

Comment:
@prithviraghavtm Kindly give me more context, what is our overall goal. 

This will help me understand, why we are not calling Libyang C APIs from golang instead of writing new parser. Thanks.

Reply:
Agree. Reuse helps.

Reply:
@prithviraghavtm @renukamanavalan: 2 doubts?

1.) Is sonic-device-health a repo internal to MSFT or is it LNF sonic repo? 
2.) Yang models specially with 'config True' is used as Source of truth (SOT) here. Due to that we need golang parser. If these yang files resides in DC application , then we may not need yang files. SOT can be saved in generic format say sql\nosql, json, Kusto etc. Hope we consider pro and cons of those approach. Thanks.

Reply:
> @prithviraghavtm @renukamanavalan: 2 doubts?
> 
> 1.) Is sonic-device-health a repo internal to MSFT or is it LNF sonic repo? 2.) Yang models specially with 'config True' is used as Source of truth (SOT) here. Due to that we need golang parser. If these yang files resides in DC application , then we may not need yang files. SOT can be saved in generic format say sql\nosql, json, Kusto etc. Hope we consider pro and cons of those approach. Thanks.

1. Yes, internal to MSFT
2. We plan to maintain config local within this service and neither CONFIG-DB nor CLI will be used.
    Yet we need YANG for following reasons.
    1.  YANG defines 3 kinds of data. a) configurable knobs; b) The result published when an action complete c) The data an action needs from previous action's publish o/p
    2. Configurable knobs, we define with "configure=true"
    3. Result attrs - we describe them so the consumer of the published data can understand (_similar to our YANG definitions for events published -- sonic-events-*.yang_ (e.g.,[ bgp-events](https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-events-bgp.yang))
    4. Actions that need certain attrs from the result of preceding actions, defines these attrs and set leafref to the corresponding actions, to ensure that the no one accidentally removes it.
    5. The integration test will simulate scenarios for each action, read the published data and verify against the YANG schema
    6. To ensure the result does have all attrs that are expected in o/p, we may tag those explicitly may be as "publish=true"